oasis-reference/src/main/kotlin/com/empowerops/volition/ref_oasis/OptimizerEndPoint.kt

```diff
@@ -1,282 +1,164 @@
-        val update = modelService.simulations.getNamed(request.name)?.update
-
-        if (update != null) {
-            update.send(request)
-        } else {
-            throw IllegalStateException("no simulation '${request.name}' or buffer full")
-        }
-        NodeChangeConfirmDTO.newBuilder().build()
-    }
-
-    override fun updateNode(request: NodeStatusCommandOrResponseDTO, responseObserver: StreamObserver<NodeChangeConfirmDTO>) = responseObserver.consume {
-        val newNode = updateFromResponse(request)
-        modelService.updateSim(newNode)
-        eventBus.post(StatusUpdateEvent("${request.name} updated"))
-        NodeChangeConfirmDTO.newBuilder().setMessage("Simulation updated with inputs: ${newNode.inputs} outputs: ${newNode.outputs}").build()
+        modelService.simulations.getNamed(request.name)?.update.sendAndRespond(request) { NodeChangeConfirmDTO.newBuilder().build() }
     }
 
-
-    override fun sendMessage(request: MessageCommandDTO, responseObserver: StreamObserver<MessageReponseDTO>) = responseObserver.consume {
-        eventBus.post(NewMessageEvent(Message(request.name, request.message)))
-        MessageReponseDTO.newBuilder().build()
-    }
-
-    override fun unregister(request: UnRegistrationRequestDTO, responseObserver: StreamObserver<UnRegistrationResponseDTO>) = responseObserver.consume {
-        val unregistered = modelService.closeSim(request.name)
-        UnRegistrationResponseDTO.newBuilder().setUnregistered(unregistered).build()
+    private suspend fun <T, U> Channel<T>?.sendAndRespond(request : T, responseBuilder : () -> U) : U {
+        if(this == null)throw IllegalStateException("no outChannel available for '$request' or buffer full")
+        send(request)
+        return responseBuilder.invoke()
     }
 
-    private fun updateFromResponse(request: NodeStatusCommandOrResponseDTO): Simulation {
-        return modelService.simulations.single { it.name == request.name }.copy(
-                name = request.name,
-                inputs = request.inputsList.map { Input(it.name, it.lowerBound, it.upperBound, it.currentValue) },
-                outputs = request.outputsList.map { Output(it.name) },
-                description = request.description
+    private fun NodeStatusCommandOrResponseDTO.generateUpdatedSimulation(): Simulation {
+        return modelService.simulations.single { it.name == name }.copy(
+                name = name,
+                inputs = inputsList.map { Input(it.name, it.lowerBound, it.upperBound, it.currentValue) },
+                outputs = outputsList.map { Output(it.name) },
+                description = description
         )
     }
 
+    private fun buildNameChangeMessage(changed: Boolean, oldName: String, newName: String) = "Name change request from $oldName to $newName ${if (changed) "succeed" else "failed"}"
```

Comment:
I am on the fence where to put these, I felt these message are earlier to compose in the the model service.